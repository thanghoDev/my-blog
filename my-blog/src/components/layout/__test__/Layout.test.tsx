import React, { ReactNode } from 'react';
import { render } from '@testing-library/react';
import { useRouter } from 'next/router';

import Layout from '..';
jest.mock('next/router', () => ({
  __esModule: true,
  useRouter: jest.fn(),
}));

describe('testing layout components', () => {
  const mockRouter = {
    isFallback: false,
    isReady: true,
    query: { search: 'search' },
  };
  (useRouter as jest.Mock).mockReturnValue(mockRouter);
  test('render layout component', () => {
    const children: ReactNode = (
      <div>
        <h1>this is header</h1>
        <h2>this title</h2>
      </div>
    );
    const { container } = render(<Layout>{children} </Layout>);
    expect(container).toMatchSnapshot();
  });
});
