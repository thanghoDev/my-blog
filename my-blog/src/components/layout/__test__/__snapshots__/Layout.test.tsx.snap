// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing layout components render layout component 1`] = `
<div>
  <div
    class="container"
  >
    <nav
      class="mt-3 navbar navbar-expand navbar-light"
    >
      <div
        class="container-fluid"
      >
        <a
          class="text-capitalize fs-2 navbar-brand"
          href="/"
        >
          meranda
        </a>
        <div
          class="offcanvas offcanvas-end"
        >
          <div
            class="offcanvas-body"
          >
            <div
              class="justify-content-end flex-grow-1 pe-3 gap-2 navbar-nav"
            >
              <a
                class="nav-link"
                data-rr-ui-event-key="#"
                href="#"
                role="button"
                tabindex="0"
              >
                <i
                  class="bi bi-facebook"
                />
              </a>
              <a
                class="nav-link"
                data-rr-ui-event-key="#"
                href="#"
                role="button"
                tabindex="0"
              >
                <i
                  class="bi bi-twitter"
                />
              </a>
              <a
                class="nav-link"
                data-rr-ui-event-key="#"
                href="#"
                role="button"
                tabindex="0"
              >
                <i
                  class="bi bi-instagram"
                />
              </a>
            </div>
            <form
              class="d-flex position-relative"
            >
              <div>
                <input
                  aria-label="Search"
                  class="form-control"
                  id="validationCustom03"
                  placeholder="Search..."
                  required=""
                  type="text"
                  value="search"
                />
              </div>
              <button
                class="btn btn-outline-success border border-light rounded-5 bg-dark"
                type="submit"
              >
                <i
                  class="bi bi-search text-light px-2"
                />
              </button>
            </form>
          </div>
        </div>
      </div>
    </nav>
    <ul
      class="gap-4 mt-4 pb-5 nav"
    >
      <li
        class="nav-item"
      >
        <a
          class="text-black-50 nav-link"
          data-rr-ui-event-key="/"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="text-black-50 nav-link"
          data-rr-ui-event-key="/category/politics"
          href="/category/politics"
        >
          Politics
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="text-black-50 nav-link"
          data-rr-ui-event-key="/category/business"
          href="/category/business"
        >
          Business
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="text-black-50 nav-link"
          data-rr-ui-event-key="/category/health"
          href="/category/health"
        >
          Health
        </a>
      </li>
    </ul>
  </div>
  <main>
    <div>
      <h1>
        this is header
      </h1>
      <h2>
        this title
      </h2>
    </div>
     
  </main>
  <footer
    class="container text-center text-secondary pt-5 pb-5"
  >
    Copyright Â©2022 All rights reserved | This template is made with by
    <code
      class="red"
    >
       
      <i
        class="bi bi-heart-fill"
      />
    </code>
     
    Colorlib
  </footer>
</div>
`;
